library/Vanilla/Formatting/DateTimeFormatter.php

```diff
@@ -0,0 +1,406 @@
+<?php
+/**
+ * @author Adam Charron <adam.c@vanillaforums.com>
+ * @copyright 2009-2019 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace Vanilla\Formatting;
+
+use Garden\StaticCacheTranslationTrait;
+
+/**
+ * Formatting methods related to dates & times.
+ */
+class DateTimeFormatter {
+
+    use StaticCacheTranslationTrait;
+
+    const FORCE_FULL_FORMAT = 'force-full-datetime-format';
+
+    /** @var DateConfig */
+    private $dateConfig;
+
+    /**
+     * @param DateConfig $dateConfig
+     */
+    public function __construct(DateConfig $dateConfig) {
+        $this->dateConfig = $dateConfig;
+    }
+
+    const NULL_TIMESTAMP_DEFALT_VALUE = '-';
+
+    /**
+     * Format a MySQL DateTime string in the specified format.
+     *
+     * @link http://us.php.net/manual/en/function.strftime.php
+     *
+     * @param string|number $timestamp A timestamp or string in Mysql DateTime format. ie. YYYY-MM-DD HH:MM:SS
+     * @param bool $isHtml Whether or not to output this as an HTML string.
+     * @param string $format The format string to use. Defaults to the application's default format.
+     * @return string
+     */
+    public function formatDate($timestamp = '', bool $isHtml = false, string $format = ''): string {
+        // Was a mysqldatetime passed?
+        if ($timestamp !== null && !is_numeric($timestamp)) {
+            $timestamp = self::dateTimeToTimeStamp($timestamp);
+        }
+
+        if ($timestamp === null) {
+            return self::t('Null Da' .
+                'te', self::NULL_TIMESTAMP_DEFALT_VALUE);
```

Comment:
what `Null Da te` string means?

Reply:
It was the existing translation string being used. 

See https://github.com/vanilla/vanilla/pull/9172/files#diff-b4fbc949423c122721c207f9d731c9d3L427

The line break looks it was some formatting mistake so I've fixed that.